[cDAC] Multi data-descriptor proposal (#118126)
dotnet/runtime: docs/design/datacontracts/data_descriptor.md

natural encoding.  Pointer values need not be aligned and need not point to addressable target
memory.

## Sub-descriptor descriptors

Each sub-descriptor descriptor is effectively a global with a type of `pointer`. They will consist of:
* a name
* a pointer value

> If these values are non-null, the pointer represents another JSON data descriptor with the specification described in this document.

Reviewer: ```suggestion
If the value is non-null, the pointer points to another [contract descriptor](contract-descriptor.md#contract-descriptor-1).
```
I assume it will point to `DotNetRuntimeContractDescriptor` that speced in the other doc, not directly to JSON.

Reviewer: Also, we use `DotNetRuntimeContractDescriptor` for both the main contract export and the data structure that it points to. We may want to rename the data structure (e.g. to just `ContractDescriptor`) to avoid confusion now that there will be multiple instances of it in the system.

> When parsing a data descriptor with sub-descriptors each sub-descriptor should be parsed then its type, global, and contract values should be merged in. If any conflicts arise when merging in sub-descriptor data, this is an error and behavior is undefined.

Reviewer: > If any conflicts arise when merging in sub-descriptor data, this is an error and behavior is undefined.

This design means that the components involved need to be aware of each other to avoid conflicts. Just pointing it out.
